### PR TITLE
Tlportal 358 - filter ctools dash events

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: server/ctools_direct_response.rb [hunk 1]
```diff
@@ -65,17 +69,38 @@ def extractAssignment(assignment)
...
+
+  # Make static method since it requires only local data and that's easier for testing.
+  def self.filter(assignment)
+
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: [#{assignment.inspect}]"
+    calendarItem = assignment['calendarItem']
+    return nil if calendarItem.nil?
+
+    calendarTimeLabelKey = calendarItem['calendarTimeLabelKey']
+    return nil if calendarTimeLabelKey.nil?
```
Comment (lsloan) [inline server/ctools_direct_response.rb]: Is it necessary to test calendarTimeLabelKey for nil *and* equality to "assignment.due.date"?  Is there a test to check for exactly "assignment.due.date" and return false if it's not that value or it's nil?

If there's not a convenient single test, do both of those tests need to be in separate conditional statements?  Couldn't they could be combined?

**Comment by lsloan** [comment]
:shipit:

File: server/ctools_direct_response.rb [hunk 2]
```diff
@@ -65,17 +69,38 @@ def extractAssignment(assignment)
...
+
+    return nil unless calendarTimeLabelKey == 'assignment.due.date'
```
Comment (zqian) [inline server/ctools_direct_response.rb]: will calendarTimeLabelKey ever be null?

File: server/spec/test_ctools_direct_response.rb [hunk 3]
```diff
@@ -64,13 +64,100 @@ def test_get_ctools_collection_todolms
...
 
-    assert_equal 3, dash_format.length, 'got multiple assignments'
+    logger.debug "#{__method__}: #{__LINE__}: dash_format: "+dash_format.inspect
+    assert_equal 2, dash_format.length, 'got multiple assignments, one filtered out'
     assert_equal 10, dash_format[0][:due_date_sort].length
+
+  end
+
+
+  def test_get_ctools_collection_multiple_keys_todolms
```
Comment (zqian) [inline server/spec/test_ctools_direct_response.rb]: wonder whether those test routines can be refactored with common functions?

File: server/ctools_direct_response.rb (hunk 2 above)
Comment (dlhaines) [inline server/ctools_direct_response.rb]: I don't have a guarantee what it may be or even if it will be present.

**Pull request merged**